### ISSUE TITLE
[BUG] AttributeError: 'numpy.float32' object has no attribute 'is_integer'

### ISSUE THREAD

**Issue by felipedram** [pycaret/pycaret#2318]
Hi!

I'm getting an `AttributeError: 'numpy.float32' object has no attribute 'is_integer'` in this part of the code:
https://github.com/pycaret/pycaret/blob/a2436a1485b325e7f99f57f99f8049177f26a8c9/pycaret/internal/preprocess.py#L194

And it makes sense because if you convert all the `float64` columns into `float32` ones
https://github.com/pycaret/pycaret/blob/a2436a1485b325e7f99f57f99f8049177f26a8c9/pycaret/internal/preprocess.py#L188-L189
, the data really can't be integer.
Moreover, I could not find why this step is taken even if I pass `preprocess=False`.

Doubts, suggestions and criticisms are most welcome.

Regards,
Felipe.

**Comment by ngupta23** [comment]
Can you paste your entire code to reproduce this error (with data if possible)?

**Comment by felipedram** [comment]
I've just got an additional info: the error just happens when the dataframe is sparse
```
encoded_ord.info()
  <class 'pandas.core.frame.DataFrame'>
  RangeIndex: 114702 entries, 0 to 114701
  Columns: 1502 entries, acarbose to IC_HOSP
  dtypes: Sparse[float32, 0.0](1502)
  memory usage: 13.1 MB
```
which is the result from

```
ord_encoder = ColumnTransformer(
    [
        ('encode_orders', TfidfVectorizer(max_features=1500, token_pattern=r"(?<=!)(.*?)(?=!)"), "DESC_ORD")
    ],
    remainder='passthrough',
)
encoded_ord = ord_encoder.fit_transform(train_data)
```
.
The env setup is
```
s = setup(
    session_id=RANDOM_SEED,
    data=encoded_ord ,
    ignore_features=[ID_COL],
    target=LABEL_COL,
    preprocess=False,
    feature_selection=True,
    fix_imbalance=True,
    data_split_stratify=True,
)
```
.
If I pass `sparse_threshold=0`, the resulting dataset is dense
```
encoded_ord.info()
  <class 'pandas.core.frame.DataFrame'>
  RangeIndex: 114702 entries, 0 to 114701
  Columns: 1502 entries, .amostra_anatomopatológica to IC_INTERNACAO
  dtypes: object(1502)
  memory usage: 1.3+ GB
```
.
It costs me 1000x more memory, but the error does not happen.

**Comment by tvdboom** [comment]
Hi, thanks for noticing the bug. This bug is already fixed in the development branch and will be released with pycaret v3.0

**Comment by felipedram** [comment]
Looking forward the version 3.0!
Anxiously!

Many tks!
